### PR TITLE
Make JSONValue.opBinaryRight!"in" inout 

### PR DESCRIPTION
**Pull request by giacomo618**

### CODE REVIEW

**Comment by thewilsonator** [comment]
Please add either a bugzilla issue or a changelog entry for this.

**Comment by CyberShadow** [comment]
@thewilsonator Please stop asking people to add changelog entries for trivial changes. Documenting this change in the changelog is a net negative.

**Comment by thewilsonator** [comment]
Hence why I asked for a bugzilla report or a changelog

**Comment by CyberShadow** [comment]
@thewilsonator What do you hope to achieve by deliberately misinterpreting my request?

You can see that the submitter is a first-time contributor. You could have guided them towards the appropriate solution for this case as opposed to doing the minimum to ensure the necessary checkboxes are checked.

**Comment by CyberShadow** [comment]
I think we can improve our tooling to introduce the notion of self-documenting pull requests OSLT, for trivial-yet-valuable pull requests with user-visible changes which don't need either a Bugzilla issue or changelog entry. The title of the pull request would be used in the generated changelog. Maybe a label?

**Review by CyberShadow — APPROVED** [review]
Code change LGTM. `opBinaryRight` uses `objectNoRef`, but that's OK because for any non-null return value, the object will provably be non-null.

I really, really don't like how we're mixing trivia with big and important changes in our changelog: it degrades the utility of the entire changelog. So, for the time being, could you please file an enhancement request against std.json in [our Bugzilla](https://issues.dlang.org/)? You can just post your test case there and a link to the PR as the description.

File: changelog/json_inout_opBinaryRight.dd [hunk 1]
```diff
@@ -0,0 +1,10 @@
+`opBinaryRight(string op : "in")` now returns `inout(JSONValue)*`
+
+`opBinaryRight(string op : "in")` returns `inout(JSONValue)*` instead
+of `const(JSONValue)*`. This makes updating a value when present easier:
+
+```
+JSONValue j = [ "foo": "bar" ];
+if (JSONValue* f = "foo" in j)
+	*f = "baz";
+```
```
Comment (ljmf00) [inline changelog/json_inout_opBinaryRight.dd:10]: A new line at the end can be added here to comply with POSIX 3.206 Line.

**Pull request merged**